### PR TITLE
F#src 3432 test integrate new driver arms hands

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

Please explain the changes you made. Add pictures or videos to explain them better if appropiate.

## Types of changes

What types of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

Put an x in the boxes that apply. You can also fill these out after creating the PR. This is a reminder of what we should look for before merging this code. I have:

- [ ] Read and follow the [contributing guidelines](https://github.com/shadow-robot/sr_documentation/blob/master/CONTRIBUTING.md).
- [ ] Checked that all tests pass with my changes
- [ ] Added tests (automatic or manual) that prove the fix is effective or that the feature works
- [ ] Added necessary documentation (if appropriate)
- [ ] Added the corresponding license to each file and add the current year of any one that you modified.
- [ ] Tested on real hardware (if appropriate)
￼

### CODE REVIEW

File: sr_robot_launch/launch/sr_hardware_control_loop.launch [hunk 1]
```diff
@@ -124,9 +183,9 @@
       </group>
       <rosparam unless="$(arg hand_ctrl)" command="load" file="$(find sr_ur_launch)/config/ra_trajectory_controller.yaml"/>
       <node if="$(arg hand_ctrl)" name="arm_trajectory_controller_spawner" pkg="controller_manager" type="spawner" output="screen"
-            args="--wait-for=/$(arg arm_id)_arm_ready --wait-for=/calibrated $(arg arm_id)_trajectory_controller"/>
+            args="--wait-for=/$(arg arm_id)_sr_ur_robot_hw/robot_program_running $(arg arm_id)_trajectory_controller"/>
```
Comment (toliver) [inline sr_robot_launch/launch/sr_hardware_control_loop.launch:193]: This will actually be an issue and it can't be merged.
We might need to modify the spawner code so that it accepts 2 topics to `wait-for`.
Remember that passing the argument twice, as we were doing, actually doesn't work (the second argument overwrites the first).
We could for example modify the spawner to allow for a string of comma-separated topics.
It would still have to work with a single topic, of course.

File: sr_robot_launch/launch/sr_hardware_control_loop.launch [hunk 2]
```diff
@@ -40,6 +40,7 @@
   <!-- Set to true to spawn the position controllers for the arm-->
   <arg name="arm_position" default="$(eval not arm_trajectory)"/>
   <arg name="arm_speed_scale" default="0.5"/>
+  <arg name="kinematics_config" default="$(find sr_ur_calibration)/calibrations/bristol_ur10_calibration.yaml"/>
```
Comment (toliver) [inline sr_robot_launch/launch/sr_hardware_control_loop.launch]: I think we don't want to introduce dependencies on sr_ur_calibration.
Shouldn't this be pointing to the default calibration?

**Review by toliver — CHANGES_REQUESTED** [review]

File: sr_robot_launch/config/la_trajectory_controller.yaml [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
-  joints:
+hardware_interface:
+  joints: &robot_joints
     - la_shoulder_pan_joint
     - la_shoulder_lift_joint
     - la_elbow_joint
     - la_wrist_1_joint
     - la_wrist_2_joint
     - la_wrist_3_joint
-    - lh_WRJ1
```
Comment (biofotis) [inline sr_robot_launch/config/la_trajectory_controller.yaml:10]: Why did you remove these joints?

File: sr_robot_launch/config/ra_trajectory_controller.yaml [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
-  joints:
+hardware_interface:
+  joints: &robot_joints
     - ra_shoulder_pan_joint
     - ra_shoulder_lift_joint
     - ra_elbow_joint
     - ra_wrist_1_joint
     - ra_wrist_2_joint
     - ra_wrist_3_joint
-    - rh_WRJ1
```
Comment (biofotis) [inline sr_robot_launch/config/ra_trajectory_controller.yaml:10]: here as well

File: sr_robot_launch/launch/sr_hardware_control_loop.launch (hunk 1 above)
Comment (carebare47) [reply sr_robot_launch/launch/sr_hardware_control_loop.launch:193]: Is this still the case since we'll be modifying joint_state_publisher.launch?

File: sr_robot_launch/launch/sr_hardware_control_loop.launch (hunk 2 above)
Comment (carebare47) [reply sr_robot_launch/launch/sr_hardware_control_loop.launch]: Yes, good spot https://github.com/shadow-robot/sr_interface/pull/446/commits/dfc48c2ad7b41e83a4c615b3f234232e4b8d25a3

File: sr_robot_launch/config/la_trajectory_controller.yaml (hunk 3 above)
Comment (carebare47) [reply sr_robot_launch/config/la_trajectory_controller.yaml:10]: Good question, should be fixed now
